tests/testutils/telemetry/traces_test.go

```diff
@@ -0,0 +1,273 @@
+	require.Equal(t, expectedResourceTraces, FlattenResourceTraces(resourceTraces))
+}
+
+func TestFlattenResourceTracesByScopeSpansIdentity(t *testing.T) {
+	resource := Resource{Attributes: &map[string]any{"attribute_three": true, "attribute_four": 23456}}
+	ss := ScopeSpans{Scope: InstrumentationScope{
+		Name: "an instrumentation library", Version: "an instrumentation library version",
+	}, Spans: []Span{}}
+	resourceTraces := ResourceTraces{
+		ResourceSpans: []ResourceSpans{
+			{Resource: resource, ScopeSpans: []ScopeSpans{}},
+			{Resource: resource, ScopeSpans: []ScopeSpans{ss}},
+			{Resource: resource, ScopeSpans: []ScopeSpans{ss, ss}},
+			{Resource: resource, ScopeSpans: []ScopeSpans{ss, ss, ss}},
+		},
+	}
+	expectedResourceTraces := ResourceTraces{
+		ResourceSpans: []ResourceSpans{
+			{Resource: resource, ScopeSpans: []ScopeSpans{ss}},
+		},
+	}
+	require.Equal(t, expectedResourceTraces, FlattenResourceTraces(resourceTraces))
+}
+
+func TestFlattenResourceTracesBySpansIdentity(t *testing.T) {
+	resource := Resource{}
+	spans := []Span{
+		{Name: "a span"},
+		{Name: "another span", Attributes: &map[string]any{}},
+		{Name: "yet another span", Attributes: &map[string]any{"attr": "value"}},
+	}
+	ss := ScopeSpans{Spans: spans}
+	resourceTraces := ResourceTraces{
+		ResourceSpans: []ResourceSpans{
+			{Resource: resource, ScopeSpans: []ScopeSpans{}},
+			{Resource: resource, ScopeSpans: []ScopeSpans{{Spans: spans[0:1]}}},
+			{Resource: resource, ScopeSpans: []ScopeSpans{{Spans: spans[1:3]}}},
+		},
+	}
+	expectedResourceTraces := ResourceTraces{
+		ResourceSpans: []ResourceSpans{
+			{Resource: resource, ScopeSpans: []ScopeSpans{ss}},
+		},
+	}
+	require.Equal(t, expectedResourceTraces, FlattenResourceTraces(resourceTraces))
+}
+
+func TestTraceContainsAllSelfCheck(t *testing.T) {
+	resourceTraces := loadedResourceTraces(t)
+	containsAll, err := resourceTraces.ContainsAll(resourceTraces)
+	require.True(t, containsAll)
+	require.NoError(t, err)
+}
+
+func TestTraceContainsAllNoBijection(t *testing.T) {
+	received := loadedResourceTraces(t)
+
+	expected, err := LoadResourceTraces(filepath.Join(".", "testdata", "traces", "expected-traces.yaml"))
+	require.NoError(t, err)
+	require.NotNil(t, expected)
+
+	containsAll, err := received.ContainsAll(*expected)
+	require.True(t, containsAll, err)
+	require.NoError(t, err)
+
+	// Since expected-traces.yaml specify a single resource spans, they will never find resource spans from received.
+	containsAll, err = expected.ContainsAll(received)
+	require.False(t, containsAll)
+	require.Error(t, err)
+	// The function used to generate the string representation of the span doesn't guarantee the order of the fields
+	// test both possibilities to ensure the test passes in all platforms.
+	errMsg := err.Error()
+	containsExpectedMissingSpanMsg :=
+		strings.Contains(errMsg, "Missing Spans: [name: span_one\nattributes:\n  one_attr: one_value\n  two_attr: two_value\n]") ||
```

Comment:
I'm under the impression yaml marshaling is sorted/deterministic?

Reply:
At least not being Windows and macOS https://github.com/signalfx/splunk-otel-collector/actions/runs/4856638013/jobs/8656548917?pr=3051#step:8:360

Reply:
I mean the test was passing locally on my Windows machine but failed on the mac when I tried a local repro.

Reply:
Fixed via 31f0a1c0e0122ada5a7ec4dabbdca9e21444ebbd thanks for insisting on this issue.

Reply:
nice, wasn't sure if I was misreading or conflating w/ something else